openassessment/xblock/static/js/src/lms/components/WaitingStepDetailsPage.jsx

```diff
@@ -0,0 +1,117 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import {
+  Alert, Container, Row, Spinner,
+} from '@edx/paragon';
+import WaitingStepList from './WaitingStepList';
+
+const WaitingStepPage = ({ waitingStepDataUrl }) => {
+  // Abort controller to clean up API calls on unmount
+  const abortController = new AbortController();
+  const signal = abortController.signal;
+
+  const [loading, setLoading] = useState(true);
+  const [error, setError] = useState(false);
+  const [waitingStepDetails, setWaitingStepDetails] = useState([]);
+
+  const fetchWaitingStepDetails = async () => {
+    // Clear error and display loading component
+    setLoading(true);
+    setError(false);
+    try {
+      // Retrieve waiting step data and decode JSON
+      const response = await fetch(waitingStepDataUrl, { signal });
+      const json = await response.json();
+      // Set table data
+      setWaitingStepDetails(json);
+    } catch (requestError) {
+      setError(true);
+    } finally {
+      setLoading(false);
+    }
+  };
+
+  useEffect(() => {
+    fetchWaitingStepDetails();
+
+    // Abort all requests on unmount
+    return function cancel() {
+      abortController.abort();
```

Comment:
Or, you can also move this out all this code to the standard reducer-selector-thunk etc framework. However, given the small scope of this. My personal feeling is that this is a minor issue that can be ignored for now, and will actually disappear when more of ORA is built with react.

Reply:
@xitij2000 Thanks for the insights, I think that it's ok to ignore the issue for now. I've also removed the `abortController` since it wasn't actually doing anything regarding the issue.